core/systemcontractupgrade/system-contract-upgrade.go

```diff
@@ -0,0 +1,116 @@
+			},
+		},
+	}
+}
+
+func UpgradeBuildInSystemContract(config *params.ChainConfig, blockNumber *big.Int, statedb *state.StateDB) {
+	if config == nil || blockNumber == nil || statedb == nil {
+		return
+	}
+	var network string
+	switch core.GenesisHash {
+	/* Add mainnet genesis hash */
+	case params.ChapelGenesisHash:
+		network = chapel
+	case params.RialtoGenesisHash:
+		network = rialto
+	}
+
+	logger := log.New("system-contract-upgrade")
+	if config.IsOnRamanujan(blockNumber) {
+		applySystemContractUpgrade(ramanujanUpgrade[network], blockNumber, statedb, logger)
+	}
+
+	/*
+		apply other upgrades
+	*/
+}
+
+func applySystemContractUpgrade(upgrade *Upgrade, blockNumber *big.Int, statedb *state.StateDB, logger log.Logger) {
+	if upgrade == nil {
+		return
+	}
+
+	logger.Info(fmt.Sprintf("Apply upgrade %s at height %d", upgrade.UpgradeName, blockNumber.Int64()))
+	for _, cfg := range upgrade.Configs {
+		logger.Info(fmt.Sprintf("Upgrade contract %s to commit %s", cfg.ContractAddr.String(), cfg.CommitUrl))
```

Comment:
==> `		logger.Info(fmt.Sprintf("Upgrade contract  to commit ", "address", cfg.ContractAddr.String(),"commit url", cfg.CommitUrl))
`

Reply:
`commit url` contains space. Besides, I think no need to follow this style.